Assets/Scripts/GameManager.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GameManager : MonoBehaviour
+{
+    private float _currentTime = 0;
+    [SerializeField]
+    private float startingTime = 60f;
+    [SerializeField]
+    private int currentLevel = 1;
+
+    private UIManager _UIManager;
+    private ShapesManager _shapesManager;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        _UIManager = GameObject.Find("Canvas").GetComponent<UIManager>();
```

Comment:
So using `.Find()` is a *highly* discouraged practice, especially in this case. There are multiple better ways you can approach this that are way more performant than `.Find()` and `.GetComponent()`.

1) Add the `[SerializedField]` attribute to `_UIManager` and `_ShapesManager`. Then, the game objects that have those scripts on them can be dragged into the field, giving you direct access to those instances.

2) You can make `UIManager` and `ShapesManager` [global singletons](https://gamedevbeginner.com/singletons-in-unity-the-right-way/) that are accessible via a call like `UIManager.instance`. This way you wouldn't even have to link the game objects, you could just call the instance from anywhere in the code. NOTE: the singleton route should be used in cases where only 1 of that class should ever exist throughout the lifetime of the game. (That said, you could potentially make a case for both `UIManager` and `ShapesManager` being viable singleton candidates in this context. Up to you.)